colab_repoclone/git_access.py

```diff
@@ -0,0 +1,105 @@
+import os
+from getpass import getpass
+
+
+class RepoClone:
+    def __init__(self, repo, branch="master", method="env"):
+        """
+        __init__
+
+        Gathers user's GitHub credentials and clones desired repository
+        with push/pull capabilities.
+
+        """
+
+        if method == "env":
+            self.github_key = os.getenv("GITHUB_KEY")
+            self.github_user = os.getenv("USER_NAME")
+            self.github_email = os.getenv("USER_EMAIL")
+            if None in [self.github_key, self.github_user, self.github_email]:
+                raise EnvironmentError(
+                    "To use the *env* keyword you must provide a GitHub username, \
+                    email and access token in your vars.env file"
+                )
+        else:
+            self.get_access()
+
+        repo_url = repo.split("//")[1]
+        self.repo_name = repo.split("/")[-1].replace(".git", "")
+        self.access_repo = "https://{GITHUB_USER}:{GITHUB_KEY}@{REPO_URL}".format(
+            GITHUB_USER=self.github_user, GITHUB_KEY=self.github_key, REPO_URL=repo_url
+        )
+        self.branch = branch
+
+        self.clone()
+
+    def clone(self):
+        """
+        clone
+
+        Clones repository into Google Colab environment using username and
+        access key to provide push/pull capabilities.
+
+        """
+
+        if self.branch == "master":
+            clone_cmd = "git clone {ACCESS_REPO}".format(ACCESS_REPO=self.access_repo)
+        else:
+            clone_cmd = "git clone --branch {BRANCH} {ACCESS_REPO}".format(
+                BRANCH=self.branch, ACCESS_REPO=self.access_repo
+            )
+
+        os.system(clone_cmd)
```

Comment:
I have a hunch that you're going to need to activate the shell here

```suggestion
        os.system(clone_cmd, shell=True)
```

(and below)

Reply:
do I need this on ever command? I've done so, but then also made the change you suggested to using subprocess.call - do I still need it?